### PR TITLE
thunderbird: configure signature if set

### PR DESCRIPTION
**Pull request by Lykos153**
### Description

The identity's signature  will be configured according to `signature.text` and `signature.showSignature`. Signature commands are not supported by Thunderbird.

### Checklist

- [x] Change is backwards compatible.

- [x] Code formatted with `./format`.

- [x] Code tested through `nix-shell --pure tests -A run.all` or `nix develop --ignore-environment .#all` using Flakes.

- [x] Test cases updated/added. See [example](https://github.com/nix-community/home-manager/commit/f3fbb50b68df20da47f9b0def5607857fcc0d021#diff-b61a6d542f9036550ba9c401c80f00ef).

- [x] Commit messages are formatted like

    ```
    {component}: {description}

    {long description}
    ```

    See [CONTRIBUTING](https://github.com/nix-community/home-manager/blob/master/docs/contributing.adoc#sec-commit-style) for more information and [recent commit messages](https://github.com/nix-community/home-manager/commits/master) for examples.

- If this PR adds a new module

  - [ ] Added myself as module maintainer. See [example](https://github.com/nix-community/home-manager/blob/068ff76a10e95820f886ac46957edcff4e44621d/modules/programs/lesspipe.nix#L6).

#### Maintainer CC
@d-dervishi @jkarlson

### CODE REVIEW

File: modules/programs/thunderbird.nix [hunk 1]
```diff
@@ -49,6 +49,11 @@ let
       "mail.identity.id_${id}.fullName" = account.realName;
       "mail.identity.id_${id}.useremail" = address;
       "mail.identity.id_${id}.valid" = true;
+      "mail.identity.id_${id}.htmlSigText" =
+        if account.signature.showSignature == "none" then
+          ""
```
Comment (teto) [inline modules/programs/thunderbird.nix:54]: are we sure this is the correct value and not something like `false` ? I couldn't find a schema of the config

**Comment by Lykos153** [comment]
https://github.com/nix-community/home-manager/blob/51e44a13acea71b36245e8bd8c7db53e0a3e61ee/modules/accounts/email.nix#L70 says it's an enumeration:

```
type = types.enum [ "append" "attach" "none" ];
```

File: modules/programs/thunderbird.nix (hunk 1 above)
Comment (teto) [reply modules/programs/thunderbird.nix:54]: I was thinking of the thunderbird config

Comment (Lykos153) [reply modules/programs/thunderbird.nix:54]: Or do you mean the Thunderbird config `htmlSigText`? That's a string. It's what is inside the text field in the identity config window. And `""` makes that field empty.

**Pull request merged**